structured-query/documentation/2021_01_29StructeredQueriesDocumentation(Draft).md

```diff
@@ -0,0 +1,132 @@
+# Structured Queries
+
+CODEX front-end allows users to create requests based on inclusion and exclusion criteria. The different inclusion and exclusion criteria are conjuncted with the "AND" and "OR" operators respectively. Resulting in a conjunctive normal form (CNF) for inclusion and disjunctive normal form (DNF) for the exclusion  criteria.
+
+The front-end created queries need to be transmitted to different back-end services which translate the CNF and DNF to queries. As common data exchange format the so called *Structured Queries* are defined.
+
+## Query Message
+
+At its core each *Structured Query* has a metadata header and the query body holding the structured query.
+
+### Query Metadata
+
+The metadata provides some basic information about the query:
+
+| Name      | Description                            |
+| :-------- | -------------------------------------- |
+| Version   | API Version                            |
+| Timestamp | Timestamp when the query was send      |
+| queryID   | unique ID to identify a specific query |
+
+## Query
+
+As previously introduced the query is based on inclusion and exclusion criteria represented in CNF and DNF. 
+
+With in the query both CNF and DNF are conjuncted with an "AND NOT" operator. 
+
+For the normal Form different building blocks are provided to represent the conjunctions of criteria.
+
+## inclusionCriteria\[][]
+
+Given a CNF:  A and (B or C)
+
+The inclusionCriteria is an array of arrays of criteria. Within the outer array all elements are conjuncted with "AND". Based on the example {A} {B or C}.
+
+Within the inner array all elements are disjuncted with "OR". Based on the example {B}{C}. A special case is given if only one element is given e.g. {A}. In this case no logic operation is applied to this element in this hierarchic element.
+
+-> inclusionCriteria= {{A}, {B, C}}
+
+## exclusionCriteria\[][]
+
+Analog for the exclusion criteria:
+
+Given a DNF:  A or (B and C)
+
+The exclusionCriteria is an array of arrays of criteria. Within the outer array all elements are disjuncted with "OR". Based on the example {A} {B and C} .
+
+Within the inner array all elements are conjuncted with "And". Based on the example {B}{C} . A special case is given if only one element is given e.g. {A}. In this case no logic operation is applied to this element in this hierarchic element.
+
+-> exclusionCriteria = {{A},  {B, C}}
+
+## criterion
+
+The previous introduced elements {A}, {B} and {C} are representative for different criteria. 
+
+Each criterion represents a unique concept or statement which is identified by its termCode.  Which can be further specified by applying a value filter.
+
+## termCode
+
+The termCode defines a concept based on a coding system (i.e. LOINC). The triplet of code, system and version identify the concept. An additional display element allows for a human readable form (This value SHALL be the same as the value define by the coding system)
```

Comment:
Please put that descriptions also in the JSON schema itself.

Reply:
Accepted.